### PR TITLE
Write perf results to diagnostics

### PR DESCRIPTION
**Pull request by rowanmiller**
This allows us to switch on diagnostics and see the perf results in VS
output window or on the command line. This makes it easy to use perf
tests as a harness when working on perf improvements.
Also adding config files so that the diagnostic messages can be easily
switched on.

### CODE REVIEW

File: test/EntityFramework7.Microbenchmarks.Core/BenchmarkRunSummary.cs [hunk 1]
```diff
@@ -68,6 +68,15 @@ public void PopulateMetrics()
             MemoryDeltaStandardDeviation = StandardDeviation(memoryDeltas, MemoryDeltaAverage);
         }
 
+        public string GetResultString()
```
Comment (anpete) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkRunSummary.cs]: ToString?

File: test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs [hunk 2]
```diff
@@ -39,6 +40,8 @@ public class BenchmarkTestCaseRunner : XunitTestCaseRunner
         private static Lazy<string> _framwork = new Lazy<string>(() => ".NETFramework");
```
Comment (anpete) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: Typo

Comment (anpete) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: Not sure why this needs to be lazy.

**Comment by anpete** [comment]
:shipit:

File: test/EntityFramework7.Microbenchmarks.Core/BenchmarkRunSummary.cs (hunk 1 above)
Comment (rowanmiller) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkRunSummary.cs]: Yeah I went back and forwards on that. I couldn't decide... I'll swap it :smile:

File: test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs (hunk 2 above)
Comment (rowanmiller) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: Here is the full code:

```
#if DNX451 || DNXCORE50
        private static Lazy<string> _framwork = new Lazy<string>(() => 
        {
            var services = Microsoft.Framework.Runtime.Infrastructure.CallContextServiceLocator.Locator.ServiceProvider; 
            var env = (Microsoft.Framework.Runtime.IRuntimeEnvironment)services.GetService(typeof(Microsoft.Framework.Runtime.IRuntimeEnvironment)); 
            return "DNX." + env.RuntimeType;
        });
#else
        private static Lazy<string> _framwork = new Lazy<string>(() => ".NETFramework");
#endif
```

Comment (anpete) [inline test/EntityFramework7.Microbenchmarks.Core/BenchmarkTestCaseRunner.cs]: Are there code paths where .Value is not accessed?

**Pull request merged**